hugr-core/src/extension/op_def.rs

```diff
@@ -243,10 +247,11 @@ impl SignatureFunc {
     ) -> Result<Signature, SignatureError> {
         let temp: PolyFuncTypeRV; // to keep alive
         let (pf, args) = match &self {
-            SignatureFunc::PolyFuncType(custom) => {
-                custom.validate.validate(args, def, exts)?;
+            SignatureFunc::CustomValidator(custom) => {
+                custom.validate.as_ref().validate(args, def, exts)?;
```

Comment:
presumably this extra `as_ref()` is needed? Code looks very similar to what we had before.

Reply:
wasn't needed, removed